inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cmd/shell/ShellExecutorImpl.java

```diff
@@ -0,0 +1,194 @@
+        cmds.add(shellPath);
+        for (String path : paths) {
+            if (StringUtils.isBlank(path)) {
+                continue;
+            }
+            cmds.add(path);
+        }
+        String[] strings = new String[cmds.size()];
+        cmds.toArray(strings);
+        return strings;
+    }
+
+    private static String arrayToString(Object[] array, String split) {
+        if (array == null || array.length == 0) {
+            return InlongConstants.BLANK;
+        }
+        StringBuilder str = new StringBuilder();
+        for (int i = 0, length = array.length; i < length; i++) {
+            if (i != 0) {
+                str.append(split);
+            }
+            str.append(array[i]);
+        }
+        return str.toString();
+    }
+
+    private static boolean HasException(String str) {
+        for (String reg : EXCEPTION_REG) {
+            Pattern pattern = Pattern.compile(reg);
+            Matcher matcher = pattern.matcher(str);
+            if (matcher.find()) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public void asyncExec(String shellPath, String... params) {
+        AsyncShellRunnable asyncShell = new AsyncShellRunnable(shellPath, this.tracker, params);
+        Thread thread = new Thread(asyncShell);
+        thread.start();
+    }
+
+    public void syncExec(String shellPath, String... params) {
+        List<String> result = new ArrayList<String>();
+        String[] cmds = merge(shellPath, params);
+        try {
+            tracker.start();
+            Process ps = Runtime.getRuntime().exec(cmds);
+            long pid = getPid(ps);
+            tracker.setProcessId(pid);
+            BufferedReader br = new BufferedReader(new InputStreamReader(ps.getInputStream()));
+            String line;
+            boolean hasException = false;
+            while ((line = br.readLine()) != null) {
+                if (HasException(line)) {
+                    hasException = true;
+                }
+                result.add(line);
+                tracker.setRunResult(arrayToString(result.toArray(), InlongConstants.NEW_LINE));
+                tracker.lineChange(line);
+            }
+            if (hasException) {
+                tracker.lineChange("Java exception exist in output");
+                tracker.fail(-1);
+                return;
+            }
+            ps.waitFor();
+            int exitValue = ps.exitValue();
+            if (exitValue != 0) {
+                tracker.fail(exitValue);
+                return;
+            }
+            tracker.success();
+        } catch (Exception e) {
+            log.error("sync exec shell failed", e);
+            result.add(e.getMessage());
+            tracker.setRunResult(arrayToString(result.toArray(), InlongConstants.NEW_LINE));
+            tracker.lineChange(e.getMessage());
+            tracker.fail(-1);
+        }
+    }
+
+    public static class AsyncShellRunnable implements Runnable {
```

Comment:
if `asyncExec()` can be removed ,this class should be removed also

Reply:
Done.